src/app/util/af-types.h

```diff
@@ -88,6 +88,37 @@ typedef void (*EmberAfGenericClusterFunction)(void);
  */
 #define EMBER_AF_NULL_MANUFACTURER_CODE 0x0000
 
+/**
+ * @brief Type for align a 16 bits value.
+ *
+ * This structure has the same size as a pointer and is used to align
+ * a 16 bit value to the 16 least significant bits of a pointer.
+ * Note that this structure is aware of the endianess and the pointer size.
+ */
+typedef struct
+{
+#if (BIGENDIAN_CPU)
+#if (UINTPTR_MAX == UINT32_MAX)
+    uint32_t align : 16;
+#elif (UINTPTR_MAX == UINT64_MAX)
+    uint64_t align : 48;
+#endif
+#endif
+
+    /**
+     * Actual value.
+     */
+    uint16_t value;
+
+#if (!BIGENDIAN_CPU)
+#if (UINTPTR_MAX == UINT32_MAX)
+    uint32_t align : 16;
+#elif (UINTPTR_MAX == UINT64_MAX)
+    uint64_t align : 48;
+#endif
+#endif
+} EmberAfAlignValue;
```

Comment:
should this be named EmberAfAlignValue16 to specify this only works for 16 bit values?

Reply:
Indeed, it’s a more proper name.

Reply:
Resolved in new commit.